### DISCUSSION TITLE
why double click or select something in insert mode, it automatic transfer to normal mode?

### DISCUSSION THREAD

**Discussion by finviman** [JetBrains/ideavim#342] [Q&A]
how to disable it ? I think only esc can change insert mode to normal mode.

**Comment by Fengjing95** [comment]
I have the same problom; but in VSCode's Vim plugin can select something without exiting insert mode.
I don't know this is a bug or so designed.

**Comment by AlexPl292** [comment]
Hi, sorry, I missed this question somehow. This is actually how vim works, it can't have a selection with an insert mode, so it switches to visual mode. However, vim has a so-called "select" mode that is pretty similar to the usual selection in other editors. You can add `set selectmode=mouse` to activate this mode with a mouse selection.

@Fengjing95

**Reply by Fengjing95** [reply]
Thanks.
